src/components/ResourceList/components/CheckableButton/CheckableButton.scss

```diff
@@ -26,7 +26,7 @@ $chekbox-label-margin: rem(20px);
   width: 100%;
   box-shadow: var(--p-override-none, shadow(faint));
 
-  [data-buttongroup-segmented] & {
+  [data-buttongroup-segmented='true'] & {
```

Comment:
This is needed, otherwise doing `<ButtonGroup segmented={false} />` would still generate `data-buttongroup-segmented="false"` and apply the styles